### PR TITLE
Use new version of protocol to correctly serialize URIs

### PR DESCRIPTION
**Pull request by dibarbet**

### CODE REVIEW

**Review by arkalyanms — APPROVED** [review]

**Comment by arkalyanms** [comment]
this seems like a good candidate for "merge without requirements" to unblock the C# release.

File: src/Features/LanguageServer/ProtocolUnitTests/Definitions/GoToDefinitionTests.cs [hunk 1]
```diff
@@ -37,6 +37,8 @@ void M()
             await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace);
 
             var results = await RunGotoDefinitionAsync(testLspServer, testLspServer.GetLocations("caret").Single());
+            // Verify that as originally serialized, the URI had a file scheme.
+            Assert.True(results.Single().Uri.OriginalString.StartsWith("file"));
```
Comment (tmat) [inline src/Features/LanguageServer/ProtocolUnitTests/Definitions/GoToDefinitionTests.cs:41]: I don't think this is necessarily correct assertion. 
It is entirely correct to create `Uri` instance using local path: `new Uri("C:\a.cs")`. This gets internally translated to `file:///C:/a.as`, so the `AbsoluteUri` would be fine but `OriginalString` would be `C:\a.cs`.

**Review by tmat — APPROVED** [review]
LGTM, other than the assertion.

File: src/Features/LanguageServer/ProtocolUnitTests/Definitions/GoToDefinitionTests.cs (hunk 1 above)
Comment (dibarbet) [reply src/Features/LanguageServer/ProtocolUnitTests/Definitions/GoToDefinitionTests.cs:41]: This is the correct assertion for this particular test.  The go to definition result is in a file - the json string it serializes should be a file schemed URI.

The test client will then convert the server's json response into LSP types.  So the original string it uses should be the file schemed URI that it got from the server.

**Comment by dibarbet** [comment]
> this seems like a good candidate for "merge without requirements" to unblock the C# release.

I want to verify unite tests pass here at least, so will let it run

**Pull request merged**